dask/tests/test_spark_compat.py

```diff
@@ -0,0 +1,108 @@
+import signal
+import sys
+import threading
+
+import pytest
+
+from dask.datasets import timeseries
+
+dd = pytest.importorskip("dask.dataframe")
+pyspark = pytest.importorskip("pyspark")
+
+from dask.dataframe.utils import assert_eq
+
+pytestmark = pytest.mark.spark
+
+# pyspark auto-converts timezones -- round-tripping timestamps is easier if
+# we set everything to UTC.
+pdf = timeseries(freq="1H").compute()
+pdf.index = pdf.index.tz_localize("UTC")
+pdf = pdf.reset_index()
+
+
+@pytest.fixture(scope="module")
+def spark_session():
+    if not sys.platform.startswith("linux"):
+        pytest.skip(
+            "Unnecessary, and hard to get spark working on non-linux platforms",
+        )
+
+    # Spark registers a global signal handler that can cause problems elsewhere
+    # in the test suite. In particular, the handler fails if the spark session
+    # is stopped (a bug in pyspark).
+    prev = signal.getsignal(signal.SIGINT)
```

Comment:
`prev` is almost certainly `None` (it is on my machine, and should be in most pytest contexts unless some weird fixture is involved).

I had to add this because `pyspark` is setting it's own handler and then not cleaning up after itself after the session is stopped. Basically, this is the cost of running pyspark in the main test suite.